qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -384,3 +389,199 @@ def run(self, run_input: List[QuantumCircuit], **options) -> FakeJob:
+        for exp_dict in list_exp_dict:
+            # skipping experiments that don't need data generation for this circuit.
+            if exp_dict["num_circuits"] <= circ_idx:
+                continue
+
+            qubits = list(exp_dict["physical_qubits"])
+            prob = self._probability_dict_to_probability_array(
+                exp_dict["prob"][circ_idx], len(qubits)
+            )
+            phase = exp_dict["phase"][circ_idx]
+            shot_num = 0
+
+            for output_number, number_of_occurrences in enumerate(
+                self._rng.multinomial(shots, prob, size=1)[0]
+            ):
+                state_str = str(format(output_number, "b").zfill(len(qubits)))
+                for _ in range(number_of_occurrences):
+                    # the iteration on the string variable state_str starts from the MSB. For
+                    # readability, we will reverse the string so the loop will run from the LSB to MSB.
+                    for qubit_idx, qubit, char_qubit in zip(
+                        range(len(qubits)), qubits, state_str[::-1]
+                    ):
+
+                        i_center = iq_centers[qubit][int(char_qubit)][0]
+                        q_center = iq_centers[qubit][int(char_qubit)][1]
+
+                        # we use 'sample_idx_shift' to take the sample corresponding to the current qubit
+                        # in 'qubits_iq_rand[shot_num]'.
+                        point_i = (
+                            i_center + qubits_iq_rand[shot_num][qubit_idx + sample_idx_shift][0]
+                        )
+                        point_q = (
+                            q_center + qubits_iq_rand[shot_num][qubit_idx + sample_idx_shift][1]
+                        )
+
+                        # Adding phase if not 0.0
+                        if not np.allclose(phase, 0.0):
+                            complex_iq = (point_i + 1.0j * point_q) * np.exp(1.0j * phase)
+                            point_i, point_q = np.real(complex_iq), np.imag(complex_iq)
+
+                        memory[shot_num].append([point_i, point_q])
+                    shot_num += 1
+
+            sample_idx_shift = sample_idx_shift + len(qubits)
+        return memory
+
+    def _parallel_generate_data(
+        self,
+        list_exp_dict: List[Dict[str, Union[List, int]]],
+        circ_idx: int,
+    ) -> Dict[str, Any]:
+        """
+        Generate data for the circuit.
+        Args:
+            list_exp_dict (List): A List of dictionaries, each dictionary contains data of an experiment.
+            circ_idx (int): The circuit number we simulate.
+
+        Returns:
+            A dictionary that's filled with the simulated data. The output format is different between
+            measurement level 1 and measurement level 2.
+
+        Raises:
+            QiskitError: if 'meas_level = MeasLevel.CLASSIFIED'.
```

Comment:
Probably not...

Reply:
what about `"QiskitError: Raising an error if classified data is requested."` ?